tutorials/tagger/basic_allennlp.py

```diff
@@ -0,0 +1,202 @@
+from allennlp.modules.seq2seq_encoders import Seq2SeqEncoder, PytorchSeq2SeqWrapper
+from allennlp.nn.util import get_text_field_mask, sequence_cross_entropy_with_logits
+
+#### We'll want to track accuracy on the training and validation datasets.
+from allennlp.training.metrics import CategoricalAccuracy
+
+#### In our training we'll need a <code>DataIterator</code>s that can intelligently batch our data.
+from allennlp.data.iterators import BucketIterator
+
+#### And we'll use AllenNLP's full-featured <code>Trainer</code>.
+from allennlp.training.trainer import Trainer
+
+#### Finally, we'll want to make predictions on new inputs, more about this below.
+from allennlp.predictors import SentenceTaggerPredictor
+
+torch.manual_seed(1)
+
+#### Our first order of business is to implement our <code>DatasetReader</code> subclass.
+class PosDatasetReader(DatasetReader):
+    """
+    DatasetReader for PoS tagging data, one sentence per line, like
+
+        The###DET dog###NN ate###V the###DET apple###NN
+    """
+    #### The only parameter our <code>DatasetReader</code> needs is a dict of <code>TokenIndexer</code>s that specify how to convert tokens into indices. By default we'll just generate a single index for each token (which we'll call "tokens") that's just a unique id for each distinct token. (This is just the standard "word to index" mapping you'd use NLP tasks.)
+    def __init__(self, token_indexers: Dict[str, TokenIndexer] = None) -> None:
+        super().__init__(lazy=False)
+        self.token_indexers = token_indexers or {"tokens": SingleIdTokenIndexer()}
+
+    #### <code>DatasetReader.text_to_instance</code> takes the inputs corresponding to a training example (in this case the tokens of the sentence and the corresponding part-of-speech tags), instantiates the corresponding <a href="https://github.com/allenai/allennlp/blob/master/tutorials/notebooks/data_pipeline.ipynb"><code>Field</code>s</a> (in this case a <code>TextField</code> for the sentence and a <code>SequenceLabelField</code> for its tags), and returns the <code>Instance</code> containing those fields. Notice that the tags are optional, since we'd like to be able to create instances from unlabeled data to make predictions on them.
+    def text_to_instance(self, tokens: List[Token], tags: List[str] = None) -> Instance:
+        sentence_field = TextField(tokens, self.token_indexers)
+        fields = {"sentence": sentence_field}
+
+        if tags:
+            label_field = SequenceLabelField(labels=tags, sequence_field=sentence_field)
+            fields["labels"] = label_field
+
+        return Instance(fields)
+
+    #### The other piece we have to implement is <code>_read</code>, which takes a filename and produces a stream of <code>Instance</code>s. Most of the work has already been done in <code>text_to_instance</code>.
+    def _read(self, file_path: str) -> Iterator[Instance]:
+        with open(file_path) as f:
```

Comment:
Did you actually try copying and pasting this code and running it?  I think you need a `cached_path` here.

Reply:
I didn't run it with the URLs because the files aren't on GitHub until I merge this PR 😢

Reply:
fixed